lib/partition/partition.c

```diff
@@ -250,7 +251,9 @@ int partition_publish(const char *device, off_t offset)
 
 				/* Some linux distros make use of subpartitions.
 				 * Scan some devices recursively to publish them. */
-				if (strcmp(name, "system") == 0 || strcmp(name, "userdata") == 0)
+				if (strcmp(name, "system") == 0
+				    || strcmp(name, "userdata") == 0
+				    || strcmp(name, "rootfs") == 0)
```

Comment:
might be worth combining this with `bootable_parts` but who knows what the future will bring!